packages/block-library/src/utils/interactivity/portals.js

```diff
@@ -0,0 +1,82 @@
+import { createElement, render } from 'preact';
+
+/**
+ * @param {import('../../src/index').RenderableProps<{ context: any }>} props
+ */
+function ContextProvider(props) {
+	this.getChildContext = () => props.context;
+	return props.children;
+}
+
+/**
+ * Portal component
+ * @this {import('./internal').Component}
+ * @param {object | null | undefined} props
+ *
+ * TODO: use createRoot() instead of fake root
+ */
+function Portal(props) {
+	const _this = this;
+	let container = props._container;
+
+	_this.componentWillUnmount = function () {
```

Comment:
if we use arrow-functions we shouldn't need to worry about `this` assignment, which I'm guessing is why we set `_this = this`?

Reply:
on the other hand we might be able to use the `class` form and not worry about it either.

Reply:
This file is copied and pasted from the [portals.js](https://github.com/preactjs/preact/blob/master/compat/src/portals.js) file in the Preact library.

Do you think we should introduce changes to this file or keep it in sync with Preact?